state/state.md

```diff
@@ -1 +1,45 @@
-# State
+# BlockExecutor
+
+## Abstract
+
+ The `BlockExecutor` is a component responsible for creating, applying, and maintaining blocks and state in the system. It interacts with the mempool and the application via the ABCI interface.
+
+## Component Description
+
+ The `BlockExecutor` is initialized with a proposer address, `namespace ID`, `chain ID`, `mempool`, `proxyApp`, `eventBus`, and `logger`. It uses these to manage the creation and application of blocks. It also validates blocks and commits them, updating the state as necessary.
+
+## Detailed Description
+
+- `NewBlockExecutor`: This method creates a new instance of `BlockExecutor`. It takes a proposer address, `namespace ID`, `chain ID`, `mempool`, `proxyApp`, `eventBus`, and `logger` as parameters.
```

Comment:
Thoughts on adding a relative link to the `block-manager.md` [file](https://github.com/rollkit/rollkit/blob/main/block/block-manager.md) here so the reader can see what each of these parameters mean?  Should look like `../block/block-manager.md`